ansible_base/lib/serializers/common.py

```diff
@@ -92,3 +95,14 @@ class Meta(CommonModelSerializer.Meta):
 class ImmutableCommonModelSerializer(AbstractCommonModelSerializer):
     class Meta(AbstractCommonModelSerializer.Meta):
         fields = AbstractCommonModelSerializer.Meta.fields + ['created', 'created_by']
+
+
+class UneditableSystemUserSerializer(CommonModelSerializer):
+    """
+    Disallows editing of system user (settings.SYSTEM_USERNAME).
+    Make sure to call super if you override validate(..).
+    """
+
+    def validate(self, data):
+        if data["username"] == settings.SYSTEM_USERNAME:
```

Comment:
This would not prevent, say, a PATCH to the system user to change their username to something other than the system username. You also need to check if `self.instance.username` is the system username.

Reply:
Thanks!  I'd already noticed it wasn't working in all cases after testing some things.  You clued me in to self.instance though, so I think comparing the IDs there will be more robust.  Please let me know if you see issues w/ the new method..